### ISSUE TITLE
Language extension

### ISSUE THREAD

**Issue by jw3** [jw3/example-daffodil-vscode#42]
We use the built-in XML language currently, which allows the DFDL XSD to be debugged, but also applies to other XML, like an infoset file.

Adding a DFDL extension should give ability to express schemas more precisely and will also support editing and other needs.

**Comment by mbeckerle** [comment]
By convention the extension for DFDL schemas is ".dfdl.xsd". Can that be made to work? i.e., be recognized and distinguished from just ".xsd"?

If not, I am not opposed to introducing ".dfdl" extension, we just lose the fact that they a DFDL schema is an XSD for people who are using "ordinary XSD tooling".

**Comment by jw3** [comment]
@mbeckerle yes, associating `*.dfd.xsd` to the DFDL language should be possible through a combination of language extension mentioned in this issue and configuration settings, both of which would be delivered with our Daffodil extension.

Currently our extension uses the built in XML language, which handles Daffodil schemas, but could also grab other non-dfdl files.

- https://code.visualstudio.com/api/language-extensions/overview
- https://code.visualstudio.com/docs/languages/overview#_changing-the-language-for-the-selected-file
